docs/glossary.rst

```diff
@@ -0,0 +1,102 @@
+
+    Open Source Software
+        Software with source code that anyone can inspect, modify, and enhance.
+
+    Pipeline
+        A Rasa bot's NLU system is defined by a pipeline, which is a list of NLU components in a particular order. A user input is processed by each component one by one before finally giving out the structured output.
+
+    Policy
+        Policies make decisions on how conversation flow should proceed. A Core model can have multiple policies included, and the policy whose prediction has the highest confidence decides the next action to be taken.
+
+    Rasa Core
+        The dialogue engine that decides on what to do next in a conversation.
+
+    Rasa Core policy
+        Decides which action to take at every step in the conversation. There are different policies to choose from, and you can include multiple policies in a single agent. At every turn, the policy which predicts the next action with the highest confidence will be used. If two policies predict with equal confidence, the policy with the higher priority will be used.
+
+    Rasa NLU component
+        Allows you to customize your model and finetune it on your dataset. Incoming messages are processed by a sequence of components called a pipeline. These components are executed one after another in a so-called processing pipeline. There are components for entity extraction, for intent classification, pre-processing, and others.
+
+    Repository
+        A folder for your project. Your project's repository contains all of your project's files and stores each file's revision history. You can also discuss and manage your project's work within the repository.
+
+    Skill
+        Small reusable parts of stories (e.g. setting a reminder, filling out a form, answering FAQs).
+
+    Slots
+        Memory of the bot, used to store structured information (e.g. extracted entities or results from API calls).
+
+    spaCy
+        A free, open-source library for advanced Natural Language Processing (NLP) in Python.
+
+    Story
+        Labeled conversation - an annotated / labeled conversation (so this includes the the intent / entities of the user as well as the sequence of actions and slots set) - stories can span across multiple skills.
+
+    Template / Response
+        An utterance is defined by a template, which could contain text, images, buttons, and other attachments.
+
+    Training
+        Using labeled data to train a model.
+
+    User goal
+        A goal that a user wants to achieve - this can also be ambiguous / unclear in the beginning (e.g. buying an insurance policy) - to achieve a user goal, you might need multiple skills.
+
+    Utterance
+        A type of bot action that only involves sending a pre-defined message back to the user (without running any other code).
+
+    Word embedding
+        A dense representation of a word often used as an input to machine learning algorithms.
```

Comment:
"A dense representation of a word" - I don't know what I should understand after reading this sentence